java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java

```diff
@@ -752,8 +752,14 @@ public void setBlob(int parameterIndex, Blob x) throws SQLException {
     }
 
     public void setClob(int parameterIndex, Clob x) throws SQLException {
-        throw new SQLFeatureNotSupportedException(
-            Constants.SQLExceptionMessages.SQL_FEATURE_NOT_SUPPORTED);
+        checkOpen();
+        if (x.length() > Integer.MAX_VALUE) {
+            throw new SQLFeatureNotSupportedException(
```

Comment:
`SQLFeatureNotSupportedException` isn't 100% the right exception but I think it's good enough. From the JavaDoc of `setClob`:
```
     * @throws SQLFeatureNotSupportedException  if the JDBC driver does not support this method
```

I think the intent is probably that if you don't support it then you always throw the error but we don't support the full contract of the method.

But I'm not committed to this. Lemme know if you want me to throw another exception and I will do so.